xbmc/addons/FontResource.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+*      Copyright (C) 2005-2017 Team XBMC
+*      http://kodi.tv
+*
+*  This Program is free software; you can redistribute it and/or modify
+*  it under the terms of the GNU General Public License as published by
+*  the Free Software Foundation; either version 2, or (at your option)
+*  any later version.
+*
+*  This Program is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+*  GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License
+*  along with Kodi; see the file COPYING.  If not, see
+*  <http://www.gnu.org/licenses/>.
+*
+*/
+#include "FontResource.h"
+
+#include "AddonManager.h"
+#include "dialogs/GUIDialogKaiToast.h"
+#include "filesystem/File.h"
+#include "filesystem/SpecialProtocol.h"
+#include "guilib/GUIWindowManager.h"
+#include "messaging/ApplicationMessenger.h"
+#include "messaging/helpers/DialogHelper.h"
+#include "ServiceBroker.h"
+#include "settings/Settings.h"
+
+using namespace XFILE;
+using namespace KODI::MESSAGING;
+
+using KODI::MESSAGING::HELPERS::DialogResponse;
+
+namespace ADDON
+{
+
+std::unique_ptr<CFontResource> CFontResource::FromExtension(CAddonInfo addonInfo, const cp_extension_t* ext)
+{
+  return std::unique_ptr<CFontResource>(new CFontResource(std::move(addonInfo)));
+}
+
+bool CFontResource::IsInUse() const
+{
+  std::string skin = CServiceBroker::GetSettings().GetString(CSettings::SETTING_LOOKANDFEEL_SKIN);
+  const auto& deps = CServiceBroker::GetAddonMgr().GetDepsRecursive(skin);
```

Comment:
Unless there's some specific need, there is no need to implement this. Dependencies are enforced elsewhere and common to everything.